components/content-blocks/GridBlock/CarouselGrid/index.js

```diff
@@ -64,6 +65,7 @@ function CarouselGrid({
     );
   }
 
+  if (isLoading) return <Loader speed="fast" isVisible />;
```

Comment:
The useList data fetching hook was already returning a results object that included the loading state, so in a lot of cases adding the loading spinner places was just as simple as exposing that state and doing something with it.